File: packages/device_info/lib/src/flutter_config/src/models/src/app_info/app_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'app_info.freezed.dart';
part 'app_info.g.dart';

@freezed
class AppInfo with _$AppInfo {
  const AppInfo._();

  const factory AppInfo({
    required String name,
    String? iconApp,
    required String version,
    required String packageName,
    required String type,
    int? timeNotificationSent,
    int? appSize,
    int? dataSize,
    int? cacheSize,
    int? totalSizeApp,
  }) = _AppInfo;

  factory AppInfo.fromJson(Map<String, dynamic> json) =>
      _$AppInfoFromJson(json);
}

File: packages/device_info/android/src/main/kotlin/com/rocket/device/info/models/AppInfo.kt
package com.rocket.device.info.models

import com.google.gson.Gson

class AppInfo(val appInfo: HashMap<String, Any>) {
    val name:String by appInfo
    val iconApp:String by appInfo
    val version:String by appInfo
    val packageName: String by appInfo
    val type: AppType by appInfo
    val timeNotificationSent : Int by appInfo
    val appSize: Int by appInfo
    val dataSize: Int by appInfo
    val cacheSize: Int by appInfo
    val totalSizeApp : Long by appInfo

    override fun toString(): String {
        val gson = Gson()
        return gson.toJson(appInfo)
    }
}



